_pytest/skipping.py

```diff
@@ -121,11 +121,9 @@ def _istrue(self):
                 # "holder" might be a MarkInfo or a MarkDecorator; only
                 # MarkInfo keeps track of all parameters it received in an
                 # _arglist attribute
-                if hasattr(self.holder, '_arglist'):
-                    arglist = self.holder._arglist
-                else:
-                    arglist = [(self.holder.args, self.holder.kwargs)]
-                for args, kwargs in arglist:
+                marks = getattr(self.holder, '_marks', None) \
```

Comment:
Not that this is made worse by this PR but it's not nice that this uses an internal attribute of the MarkInfo.  Should markinfo not make this public somehow instead?  How about using the MarkInfo as an iterator?  I think that gives you access to all args/kwargs individually.